cf/read_write/netcdf/netcdfread.py

```diff
@@ -455,25 +506,83 @@ def _cache_data_elements(self, data, ncvar):
             variable = g["variables"].get(ncvar)
 
         # Get the required element values
-        size = variable.size
-        if size == 1:
-            value = variable[(slice(0, 1, 1),) * variable.ndim]
-            values = (value, None, value)
+        size = data.size
+        ndim = data.ndim
+
+        char = False
+        if variable.ndim == ndim + 1:
+            dtype = variable.dtype
+            if dtype is not str and dtype.kind in "SU":
+                # This variable is a netCDF classic style char array
+                # with a trailing dimension that needs to be collapsed
+                char = True
+
+        if ndim == 1:
+            # Also cache the second element for 1-d data, on the
+            # assumption that they may well be dimension coordinate
+            # data.
+            if size == 1:
+                indices = (0, -1)
+                value = variable[...]
+                values = (value, value)
+            elif size == 2:
+                indices = (0, 1, -1)
```

Comment:
But the `1` and `-1` indices are the same object for size `2` one-dimensional data, and similarly `0` and `-1` the same for size `1` as above, surely? Unless I a missing something as to why both equivalents are listed in both cases? Obviously we don't want to store effectively duplicate information.

Reply:
Ah I see. Thanks for explaining, that makes sense to me, though if you want to have a ponder on related concerns, please go ahead! I'll keep this open (unresolved) as a reminder but feel free to close it if you wish.